lib/rules/role-supports-aria-props.js

```diff
@@ -0,0 +1,101 @@
+// @ts-check
+const {aria, elementRoles, roles} = require('aria-query')
+const {getProp, getPropValue, propName} = require('jsx-ast-utils')
+const {getElementType} = require('../utils/get-element-type')
+const ObjectMap = require('../utils/object-map')
+
+// Clean-up `elementRoles` from `aria-query`
```

Comment:
`eslint-plugin-jsx-a11y` (the upstream project) [holds](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/911#issuecomment-1340125456) that, “[W]e don't want to store knowledge of WAI-ARIA in this plugin. That knowledge should be stored in `aria-query`.”

I think that’s a good architectural principle in general, but unnecessary for temporary code like this. I’m okay mixing WAI-ARIA knowledge with rule code here; since we plan to replace this rule once the upstream one is updated.